extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/OidcRecorder.java

```diff
@@ -119,16 +120,29 @@ public void handle(AsyncResult<OAuth2Auth> event) {
 
                 auth = cf.join();
                 break;
-            } catch (OIDCException ex) {
-                if (i + 1 < connectionRetryCount) {
-                    try {
-                        Thread.sleep(2000);
-                    } catch (InterruptedException iex) {
-                        // continue connecting
+            } catch (Throwable throwable) {
+                while (throwable instanceof CompletionException && throwable.getCause() != null) {
+                    throwable = throwable.getCause();
+                }
+                if (throwable instanceof OIDCException) {
+                    if (i + 1 < connectionRetryCount) {
+                        CompletableFuture.runAsync(
```

Comment:
runAsync should never be used, it uses the default fork join pool that we have no control over.

Reply:
Right, recall you were saying in context of the CL work...